fs/shm/shmfs.c

```diff
@@ -238,8 +238,13 @@ static int shmfs_truncate(FAR struct file *filep, off_t length)
           filep->f_inode->i_private = shmfs_alloc_object(length);
           if (!filep->f_inode->i_private)
             {
+              filep->f_inode->i_size = 0;
               ret = -EFAULT;
```

Comment:
I think the right error number should be -ENOMEM

Reply:
I think both return values are equally wrong:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/truncate.html

truncate assumes that it cannot fail due to oom, most likely because it assumes truncate does not actually reserve memory and instead relies on copy-on-write (which we don't have / support).

I would not change the return value.

This same topic was mentioned by the author btw, here:
https://github.com/apache/nuttx/pull/8109#discussion_r1072503741